### ISSUE TITLE
where is  teacher model？

### ISSUE THREAD

**Issue by zuowanbushiwo** [RicherMans/CED#20]
Hi @RicherMans 

In your paper, it was stated that the `Teacher model ensemble consists of two ViT-Base models and three ViT-Large models, all of which have been independently trained on AS`. 
In the download link on Zenodo, does the file "audiotransformer_base_mae_as_10s.pt" correspond to the Teacher model's ViT-Base mentioned above? 
Also, I did not find the pretrained models for the Teacher ViT-Large on Zenodo？  and Is there also no script for training the teacher models (ViT-Base and  ViT-Large  with mae) in this project?

Thanks！

**Comment by RicherMans** [comment]
Hey @zuowanbushiwo ,
thanks again for the interest!
> In the download link on Zenodo, does the file "audiotransformer_base_mae_as_10s.pt" correspond to the Teacher model's ViT-Base mentioned above?

No I haven't included any of the teachers, except for the one from [SAT](https://zenodo.org/records/7964975/files/audiotransformer_base_mAP_47_40.pt?download=1).

Sorry that I haven't included them, but for me the goal is to provide a reproducible script and let uses use their own teacher models.
Further, I want to avoid too many unnecessary questions regarding i.e., download problems, since the teachers are also quite large.
Lastly, I don't think that many people want to 100% reproduce all results provided in any paper. Like if I would use imagenet pretrained models, should one also provide scripts on how to train these models from scratch? 
I personally want to keep the code as clean and simple as possible, in order to avoid any confusion about the main aspects of the paper.

> Also, I did not find the pretrained models for the Teacher ViT-Large on Zenodo？ and Is there also no script for training the teacher models (ViT-Base and ViT-Large with mae) in this project?

I think that there are enough available training examples for training on Audioset, notable KQQ's [Panns](https://github.com/qiuqiangkong/audioset_tagging_cnn).
My code can also be adjusted for that, actually my own [PSL](https://github.com/RicherMans/PSL/blob/main/src/run.py) code should work just fine. 
Point CED is to provide a framework, where you can plug in your own teacher models and train that in CED fashion.

**Comment by zuowanbushiwo** [comment]
Hi @RicherMans 

Thank you very much for your reply, but I still have many questions.

If I do not have these teacher models (the ensemble consists of two ViT-Base models and three ViT-Large models), how do I generate my own logits/ensemble5014/full?

Since AudioSet is not an "open source" dataset, nearly everyone who downloads it has:

  1. Different names for the audio files (depending on how you save them),
  2. Different data overall due to deletions, etc.

My Local AudioSet data  contains 20550 / 22160 of the balaned training subset, 1913637 / 2041789 of the unbalanced training subset, and 18887 / 20371 of the evaluation subset.    It may be different from the dataset you are using..
  
So when I use full_audiotransformer_tiny_T_Ensemble5014_chunk10.yaml with a logitspath: logits/ensemble5014/full/chunk_10/, does this need to match my local train_data: data/full.tsv?

Certainly, I can follow the configuration in full_base_chunk10s_topk20.yaml and use a single audiotransformer_base_4740 model, but I'm concerned that the logits generated this way might differ from those in logits/ensemble5014/full.

Thanks！

**Comment by RicherMans** [comment]
Hey @zuowanbushiwo ,
> If I do not have these teacher models (the ensemble consists of two ViT-Base models and three ViT-Large models), how do I generate my own logits/ensemble5014/full?

I hope this is explained in the [README](https://github.com/RicherMans/CED?tab=readme-ov-file#training-on-your-own-data)

Specifically, the current config simply uses a single model (as an example): https://github.com/RicherMans/CED/blob/60a0f54214ea619265e16ee257925158be2675fd/logitconfig/full_base_chunk10s_topk20.yaml#L3

Just use multiple models there, passing like:
```
model: 
  -  audiotransformer_base_4740
  -  audiotransformer_base_4740
  -   audiotransformer_base_4740
```

> Since AudioSet is not an "open source" dataset, nearly everyone who downloads it has: Different names for the audio files (depending on how you save them), Different data overall due to deletions, etc. So when I use full_audiotransformer_tiny_T_Ensemble5014_chunk10.yaml with a logitspath: logits/ensemble5014/full/chunk_10/, does this need to match my local train_data: data/full.tsv?

Yes, it needs to match the data names, unfortunately. I think we had already an issue about that: https://github.com/RicherMans/CED/issues/1#issuecomment-1705810589 


> Certainly, I can follow the configuration in full_base_chunk10s_topk20.yaml and use a single audiotransformer_base_4740 model, but I'm concerned that the logits generated this way might differ from those in logits/ensemble5014/full.

Yes, they will, but my point is that you can just use the trained CED-Base model to generate logits on your dataset. That should work too, instead of training some teacher models yourself.
Or just train couple of models using different seeds on Audioset and later merge their logits.

**Comment by zuowanbushiwo** [comment]
Hi @RicherMans 

Thank you very much for your reply. I now understand a lot more. I have a question about your [SAT ](https://github.com/RicherMans/SAT)project.

If I want to train a SAT_T_stream1s model, how would I go about it?
Because When training SAT_T  is a psl_data parameter required，Can this be generated using a trained CED-Base model by changing the [chunk_length ](https://github.com/RicherMans/CED/blob/60a0f54214ea619265e16ee257925158be2675fd/utils.py#L26)to 2 seconds?

Thanks!

**Comment by RicherMans** [comment]
Hey @zuowanbushiwo ,
> Because When training SAT_T is a psl_data parameter required，Can this be generated using a trained CED-Base model by changing the [chunk_length ](https://github.com/RicherMans/CED/blob/60a0f54214ea619265e16ee257925158be2675fd/utils.py#L26)to 2 seconds?

Yeah, thats right.
